data/lua/wml/harm_unit.lua

```diff
@@ -168,11 +194,15 @@ function wml_actions.harm_unit(cfg)
 			if experience ~= false and harmer and harmer.valid
 				and wesnoth.sides.is_enemy( unit_to_harm.side, harmer.side )
 			then
-				if kill ~= false and unit_to_harm.hitpoints <= 0 then
+				if kill ~= false and unit_to_harm.hitpoints <= 0 and xp_mode.kill then
 					harmer.experience = harmer.experience + calc_xp( unit_to_harm.level )
 				else
-					unit_to_harm.experience = unit_to_harm.experience + harmer.level
-					harmer.experience = harmer.experience + wesnoth.game_config.combat_experience * unit_to_harm.level
+					if xp_mode.defend then
+						unit_to_harm.experience = unit_to_harm.experience + harmer.level
```

Comment:
@gfgtdf Why is wesnoth.game_config.combat_experience applied to the attacker, but not the defender? I'm asking you because bc89512a5c000e0546676a3d0ae0f1be20b31731 seems to the be origin of it.

Reply:
I fixed it.